lms/models/application_instance.py

```diff
@@ -50,12 +56,18 @@ def get_by_consumer_key(cls, db, consumer_key):
 
     @classmethod
     def build_from_lms_url(  # pylint:disable=too-many-arguments
-        cls, lms_url, email, developer_key, developer_secret, encryption_key=None
+        cls,
+        lms_url,
+        email,
+        developer_key,
+        developer_secret,
+        encryption_key,
+        canvas_sections_enabled,
     ):
         """Instantiate ApplicationInstance with lms_url."""
         encrypted_secret = developer_secret
         aes_iv = None
-        if encryption_key is not None and developer_secret and developer_key:
+        if developer_secret and developer_key:
```

Comment:
I made `encryption_key` non-optional, it's never not passed

Reply:
This classmethod is a bit of a horror but I opted not to get in to refactoring it any more than that